solr/core/src/java/org/apache/solr/handler/component/QueryElevationComponent.java

```diff
@@ -1590,10 +1587,10 @@ public int getSubsetCount() {
      * Returns an iterator over all the subsets that are contained by the provided set. The returned
      * iterator does not support removal.
      *
-     * @param set This set is copied to a new {@link ImmutableSortedSet} with natural ordering.
+     * @param set This set is copied to a new {@link SortedSet} with natural ordering.
      */
     public Iterator<M> findSubsetsMatching(Collection<E> set) {
-      return new MatchIterator(ImmutableSortedSet.copyOf(set));
+      return new MatchIterator(Collections.unmodifiableSortedSet(new TreeSet<>(set)));
```

Comment:
Immutability can be removed.

Reply:
Fixed in f75201354ee64df0fc739e6f128ebe0739e8b884